src/components/user/event.tsx

```diff
@@ -279,38 +280,150 @@ export function Event({
   function renderObject(object: AbstractUuid & { __typename?: string }) {
     return (
       <Link href={object.alias ?? `/${object.id}`}>
-        {hasObject(object)
+        {hasTitle(object)
           ? object.currentRevision.title
-          : getEntityStringByTypename(object.__typename, strings)}
+          : getEntityStringIncludingFolder(object)}
       </Link>
     )
   }
 
+  // Gets localized entity name (for example "Solution") and containing folder (for example "Exercises about topic X")
+  // Full example: "Solution | Exercises about topic X"
+  function getEntityStringIncludingFolder(
+    object: AbstractUuid & { __typename?: string }
+  ) {
+    const taxonomyTerms = getTaxonomyTerms(object)
+
+    if (taxonomyTerms.length == 0) {
+      return getEntityStringByTypename(object.__typename, strings)
```

Comment:
Since you use `getEntityStringByTypename` three times and in all cases you should probably do 
``` ts
const entityString =  getEntityStringByTypename(…)
```